tests/Security/Voter/HandleCommandVoterTest.php

```diff
@@ -50,22 +50,62 @@ public function testVote($attribute, $subject, $decision, $mapping, $expected)
     public function provideTestVoteData()
     {
         return [
-            // Testcase: default access is false
-            ['handle', new FakeCommand, true, [], VoterInterface::ACCESS_DENIED],
-            // Testcase: abstain when not handling a command, but using the handle attribute
-            ['handle', null, true, [], VoterInterface::ACCESS_ABSTAIN],
-            // Testcase: abstain when not handling a command and not using the handle attribute
-            ['create', null, true, [], VoterInterface::ACCESS_ABSTAIN],
-            // Testcase: abstain when not handling a command
-            ['create', new FakeCommand, true, [], VoterInterface::ACCESS_ABSTAIN],
-            // Testcase: default is unrelated to decision manager
-            ['handle', new FakeCommand, false, [], VoterInterface::ACCESS_DENIED],
-            // Testcase: deny access if decision manager returns false
-            ['handle', new FakeCommand, false, [FakeCommand::class => ['ROLE_USER']], VoterInterface::ACCESS_DENIED],
-            // Testcase: grant access if decision manager returns true and the command is in the mapping
-            ['handle', new FakeCommand, true, [FakeCommand::class => ['ROLE_USER']], VoterInterface::ACCESS_GRANTED],
-            // Testcase: deny access if the command is not in the mapping (i.e. a default deny access case)
-            ['handle', new FakeCommand, false, ['someOtherCommand' => ['ROLE_USER']], VoterInterface::ACCESS_DENIED],
+            'default access is false' => [
+                'handle',
+                new FakeCommand,
+                true,
+                [],
+                VoterInterface::ACCESS_DENIED
+            ],
+            'default is unrelated to decision manager' => [
+                'handle',
+                new FakeCommand,
+                false,
+                [],
+                VoterInterface::ACCESS_DENIED
+            ],
+            'abstain when not handling a command, but using the handle attribute' => [
+                'handle',
+                null,
+                true,
+                [],
+                VoterInterface::ACCESS_ABSTAIN
+            ],
+            'abstain when not handling a command and not using the handle attribute' => [
+                'create',
+                null,
+                true,
+                [],
+                VoterInterface::ACCESS_ABSTAIN
+            ],
+            'abstain when handling a command and not using the handle attribute' => [
+                'create',
+                new FakeCommand,
+                true,
+                [],
+                VoterInterface::ACCESS_ABSTAIN
+            ],
+            'deny access if decision manager returns false, even if the command is in the mapping' => [
+                'handle',
+                new FakeCommand,
+                false,
+                [FakeCommand::class => ['ROLE_USER']],
```

Comment:
First off, these mappings here look legit for the test fixtures but they don't actually do anything: the AccessDecisionManager is mocked so the test passes without caring that this here.

So, we should either create a better mock or remove these from the test. That could cleanup the number of test cases we have here and we can move more of this logic to the new integration tests which actually work with the mappings.